src/Login.tsx

```diff
@@ -4,21 +4,229 @@ import { TextInput } from './stories/TextInput/TextInput'
 import { Image } from './stories/Image/Image'
 import { Form } from './stories/Form/Form'
 
-import { useState } from 'react'
+import { MutableRefObject, ReactElement, useRef, useState } from 'react'
 import { NavigateFunction, useNavigate } from 'react-router-dom'
+import { getData } from './utils/restApiWrapper'
 
 import { env } from './util'
+import { useAlert, useModal } from './components/GlobalHooks'
+import { AlertProps } from './stories/Alert/Alert'
+import { ModalProps } from './stories/Modal/Modal'
 
-export const Login: React.FC = () => {
-    const [userNameEntered, setUserNameEntered] = useState('' as string)
-    const [passwordEntered, setPasswordEntered] = useState('' as string)
-    const [userpassEnabled, setUserpassEnabled] = useState(false as boolean)
+export const commonHeaders = {
+    'X-Rucio-VO': 'def',
+    'X-Rucio-AppID': 'test',
+}
+
+function Login() {
+    const [userNameEntered, setUserNameEntered] = useState('')
+    const [passwordEntered, setPasswordEntered] = useState('')
+    const [userpassEnabled, setUserpassEnabled] = useState(false)
+
+    const authType: MutableRefObject<string> = useRef('')
 
     const navigate: NavigateFunction = useNavigate()
+    const showAlert: (options: AlertProps) => Promise<void> = useAlert()
+    const showModal: (options: ModalProps) => Promise<void> = useModal()
+
+    const accountName: MutableRefObject<string> = useRef('root')
+    const accountNameProvided: MutableRefObject<boolean> = useRef(false)
+
+    const AccountInput: ReactElement = (
+        <TextInput
+            label="Account Name"
+            placeholder="Enter Account Name (optional)"
+            size="medium"
+            kind="primary"
+            onChange={(event: any) => {
+                accountName.current = event.target.value
+                accountNameProvided.current = true
+            }}
+        />
+    )
+
+    const SignInButton: ReactElement = (
+        <div className="container-login100-form-btn m-t-17">
+            <Button
+                size="large"
+                kind="primary"
+                show="block"
+                label="Sign In"
+                type="submit"
+                disabled={
+                    passwordEntered.length == 0 || userNameEntered.length == 0
+                }
+                onClick={() => {
+                    authType.current = 'userpass'
+                }}
+            />
+        </div>
+    )
 
-    const handleSubmit = (event: any) => {
+    const makeUserPassAuthFetch = (): Promise<any> => {
+        return getData('/auth/userpass', '', {
+            ...commonHeaders,
+            'X-Rucio-Username': userNameEntered,
+            'X-Rucio-Password': passwordEntered,
+            'X-Rucio-Account': accountName.current,
+        })
+    }
+
+    const fetchUserScopeToken = () => {
+        makeUserPassAuthFetch()
+            .then((data: any) => {
```

Comment:
the type any fails linting. 
```
/workspaces/webui/src/components/GlobalHooks.tsx
  16:47  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  17:69  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  18:69  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/workspaces/webui/src/components/Home.tsx
  7:21  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/workspaces/webui/src/components/Login.tsx
   41:31  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   57:44  warning  Expected '===' and instead saw '=='       eqeqeq
   57:75  warning  Expected '===' and instead saw '=='       eqeqeq
   66:47  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   77:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   84:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  110:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  111:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  112:33  warning  Expected '===' and instead saw '=='       eqeqeq
  124:51  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  130:53  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  137:55  warning  Expected '===' and instead saw '=='       eqeqeq
  139:63  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  153:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  164:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  181:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  192:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  199:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  207:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  214:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  220:40  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  223:29  warning  Expected '===' and instead saw '=='       eqeqeq
  225:36  warning  Expected '===' and instead saw '=='       eqeqeq
  308:59  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  321:59  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/workspaces/webui/src/stories/Page/Page.tsx
  11:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  14:43  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/workspaces/webui/src/stories/TextInput/TextInput.tsx
  12:23  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
```

Some of them we cannot really fix, but I am leaving a comment here so we can address them in follow ups.

Reply:
> the type any fails linting.

Should not the build ideally, as it's just a warning.